src/main/java/io/strimzi/kproxy/internal/BufferReleaseHandler.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.strimzi.kproxy.internal;
+
+import java.util.List;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelInboundHandlerAdapter;
+
+public class BufferReleaseHandler extends ChannelInboundHandlerAdapter {
```

Comment:
I won't add a specific handler just for this, because:
- [performance] handlers are nodes of the linked list of handlers and you'll pay the cost to iterated through them for each wrte/writeAndFlush/flush
- [performance] attributes lookups are costy (although I've tried to improve it some time ago -> see [DefaultAttributeMap](https://github.com/netty/netty/blob/e5951d46fc89db507ba7d2968d2ede26378f0b04/common/src/main/java/io/netty/util/DefaultAttributeMap.java#L39) for the binary search used for each lookup)
- releasing buffers is a mandatory operation for a mandatory already existing handler: better to handle release by creating a proper event/message that would traverse existing channel handlers instead

I would reuse the concept of reference counting 
eg 
[ReferenceCountedOpenSslContext](https://github.com/netty/netty/blob/626f94f069647f4e60a6863ca6540be1722e6e45/handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslContext.java#L134) or [HAProxyMessage](https://github.com/netty/netty/blob/626f94f069647f4e60a6863ca6540be1722e6e45/codec-haproxy/src/main/java/io/netty/handler/codec/haproxy/HAProxyMessage.java#L37) and create an ad-hoc reference counted event/message type wrapping `DecodedRequestFrame`including the used 'ByteBuf`s to be released by the latest channel using it (see https://netty.io/wiki/reference-counted-objects.html#who-destroys-it for an hint about how to handle it 
ie 
> If a component consumes a reference-counted object and knows nothing else will access it anymore (i.e., does not pass along a reference to yet another component), the component should destroy it. 

)